### PR TITLE
Update to 1.21

### PR DESCRIPTION
**Pull request by CommandGenius**
This is still a WIP, there are crashes when modifying settings.

### CODE REVIEW

**Comment by TexBlock** [comment]
Thank you port to 1.21. I'll wait until ObsidianUI is completely done with the 1.21 port (SOON) to port the RDL, and this PR will still be kept.

**Comment by clmc99** [comment]
I get the following error when I try to ```gradlew build```:

```
> Task :fabric:compileJava
C:\Users\clmcm\AppData\Roaming\.minecraft\mods\compiling\RyoamicLights\fabric\src\main\java\org\thinkingstudio\ryoamiclights\fabric\mixin\VideoOptionsScreenMixin.java:28: error: VideoOptionsScreenMixin is not abstract and does not override abstract method addOptions() in GameOptionsScreen
public class VideoOptionsScreenMixin extends GameOptionsScreen {
       ^
Note: C:\Users\clmcm\AppData\Roaming\.minecraft\mods\compiling\RyoamicLights\fabric\src\main\java\org\thinkingstudio\ryoamiclights\fabric\RyoamicLightsFabric.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
1 error

> Task :fabric:compileJava FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':fabric:compileJava'.
> Compilation failed; see the compiler error output for details.
```

**Comment by CommandGenius** [comment]
> I get the following error when I try to `gradlew build`:
> 
> ```
> > Task :fabric:compileJava
> C:\Users\clmcm\AppData\Roaming\.minecraft\mods\compiling\RyoamicLights\fabric\src\main\java\org\thinkingstudio\ryoamiclights\fabric\mixin\VideoOptionsScreenMixin.java:28: error: VideoOptionsScreenMixin is not abstract and does not override abstract method addOptions() in GameOptionsScreen
> public class VideoOptionsScreenMixin extends GameOptionsScreen {
>        ^
> Note: C:\Users\clmcm\AppData\Roaming\.minecraft\mods\compiling\RyoamicLights\fabric\src\main\java\org\thinkingstudio\ryoamiclights\fabric\RyoamicLightsFabric.java uses or overrides a deprecated API.
> Note: Recompile with -Xlint:deprecation for details.
> 1 error
> 
> > Task :fabric:compileJava FAILED
> 
> FAILURE: Build failed with an exception.
> 
> * What went wrong:
> Execution failed for task ':fabric:compileJava'.
> > Compilation failed; see the compiler error output for details.
> ```

try rebuilding now

**Comment by CommandGenius** [comment]
Also, it looks like the crashing is gone.

**Comment by Giovix16** [comment]
How can I download it?

**Comment by CommandGenius** [comment]
> How can I download it?

build it

**Comment by TexBlock** [comment]
@CommandGenius i released ObsidianUI 1.21 alpha version.
please use modrinth maven replace mavenLocal

**Comment by CommandGenius** [comment]
> @CommandGenius i released ObsidianUI 1.21 alpha version

i know, i updated it to stop using mavenLocal.

[also here's a prebuilt version for early adopters](https://github.com/CommandGenius/RyoamicLights/releases/download/1.21/RyoamicLights-fabric-0.2.7+mc1.21.jar)

File: settings.gradle [hunk 1]
```diff
@@ -10,6 +10,6 @@ pluginManagement {
 
 include("common")
 include("fabric")
-include("neoforge")
+//include("neoforge")
```
Comment (TexBlock) [inline settings.gradle:13]: why？

**Comment by TexBlock** [comment]
Merged in `1.21.x-architectury`

**Pull request closed without merging**